crates/ruff_python_parser/src/lexer.rs

```diff
@@ -339,14 +351,38 @@ impl<'source> Lexer<'source> {
                 let imag = f64::from_str(number.as_str()).unwrap();
                 Ok(Tok::Complex { real: 0.0, imag })
             } else {
-                let value = number.as_str().parse::<BigInt>().unwrap();
-                if start_is_zero && !value.is_zero() {
-                    // leading zeros in decimal integer literals are not permitted
-                    return Err(LexicalError {
-                        error: LexicalErrorType::OtherError("Invalid Token".to_owned()),
-                        location: self.token_range().start(),
-                    });
-                }
+                let value = match number.as_str().parse::<isize>() {
+                    Ok(value) => {
+                        if start_is_zero && value != 0 {
+                            // leading zeros in decimal integer literals are not permitted
+                            return Err(LexicalError {
+                                error: LexicalErrorType::OtherError("Invalid Token".to_owned()),
+                                location: self.token_range().start(),
+                            });
+                        }
+                        Int::Small(value)
+                    }
+                    Err(err) => {
+                        if matches!(
+                            err.kind(),
+                            IntErrorKind::PosOverflow | IntErrorKind::NegOverflow
+                        ) {
+                            if start_is_zero {
+                                // leading zeros in decimal integer literals are not permitted
+                                return Err(LexicalError {
+                                    error: LexicalErrorType::OtherError("Invalid Token".to_owned()),
+                                    location: self.token_range().start(),
+                                });
+                            }
```

Comment:
Can we add a test for a very long sequence of `0`: `000000000000000000000000000000000000000000000000000....` to ensure it gets catched by line 355?